SwiftLeeds/Views/Components/FancyHeaderView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  HeaderView.swift
+//  SwiftLeeds
+//
+//  Created by Kannan Prasad on 05/07/2022.
+//
+
+import SwiftUI
+
+struct FancyHeaderView: View {
+    let title: String
+    let foregroundImageURL: URL
+    let backgroundImageURL: URL
+    
+    private let foregroundImageWidth = 160.0
+    private let aspectRatio = 1.66
+    
+    @State var foregroudGroupViewHeight: CGFloat = .zero
+    
+    var body: some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .edgesIgnoringSafeArea(.top)
+            .aspectRatio(aspectRatio, contentMode: .fill)
+            .background(backgroundImage
+                            .aspectRatio(contentMode: .fill))
+            .overlay(foregroudGroupView,alignment: .center)
+            .padding(.bottom,foregroudGroupViewHeight/2)
+    }
+    
+    private var backgroundImage: some View {
+        AsyncImage(url: backgroundImageURL) { phase in
+            
+            switch phase {
+            case .empty:
+                ProgressView()
+            case .success(let image):
+                image
+                    .resizable()
+                    .aspectRatio(aspectRatio, contentMode: .fill)
+            case .failure(_):
+                ProgressView()
+                    .tint(.white)
+                    .opacity(0.5)
+                //TODO: Remove the loaded and set a default image
+            @unknown default:
+                ProgressView()
+                    .tint(.white)
+                    .opacity(0.5)
+            }
+        }
+    }
+    
+    private var foregroudGroupView: some View {
+        GeometryReader { geometry in
+            VStack{
+                foregroundImage
+                    .padding(.bottom, Padding.stackGap)
+                    .shadow(color: Color.black.opacity(1/3), radius: 8, x: 0, y: 0)
+                Text(title)
+                    .foregroundColor(.primary)
+                    .font(.subheadline.weight(.bold))
+                    .padding(.bottom, Padding.stackGap)
```

Comment:
I don't think this should have intrinsic padding beyond the amount needed to account for the stack - the idea with components like this is they're drop in, and the screens can use them as they please, so padding you can't opt-out of goes against that

Reply:
As per the Zeplin design there was a padding below for the title, so kept this here